### DISCUSSION TITLE
code-server with the --proxy-domain

### DISCUSSION THREAD

**Discussion by a224327780** [coder/code-server#6092] [Q&A]
- run log
```
[2023-03-21T03:41:18.044Z] info  HTTP server listening on http://0.0.0.0:7071/
[2023-03-21T03:41:18.044Z] info    - Authentication is enabled
[2023-03-21T03:41:18.044Z] info      - Using password from $PASSWORD
[2023-03-21T03:41:18.044Z] info    - Not serving HTTPS
[2023-03-21T03:41:18.044Z] info    - Proxying the following domain:
[2023-03-21T03:41:18.044Z] info      - *.code-server.xxx.xyz
```

![image](https://user-images.githubusercontent.com/78008038/226560124-4976e3bc-2b2f-47aa-89e9-77ec7ec6bb9e.png)

- Local Address https://code-server.xxx.xyz/proxy/3000/
- I hope so https://3000.code-server.xxx.xyz/
- --proxy-domain Why didn't it take effect

**Comment by code-asher** [comment]
Answered here: https://github.com/coder/code-server/issues/6091

> Yeah this is confusing, currently we default to the path-based proxy even if a proxy domain is set.
> There is an issue for it here: https://github.com/coder/code-server/issues/5311
> For now the workaround is to manually set the VSCODE_PROXY_URI environment variable to https://{{port}}.code-server.xxx.xyz

**Reply by a224327780** [reply]
```
➜  workspace echo $VSCODE_PROXY_URI
https://{{port}}.code-server.xxx.xyz
```
- Setting environment variables still doesn't work

**Reply by code-asher** [reply]
Are you setting it in a way that the main code-server process can see it?  I tested with Docker but it works:

```
docker run --rm -e 'VSCODE_PROXY_URI=https://some-{{port}}-domain.tld' -p 8080:8080 codercom/code-server --auth none
```

![proxy](https://user-images.githubusercontent.com/45609798/226987948-fad1edff-c11e-46c4-8fa5-1b7fce87b690.png)

**Reply by a224327780** [reply]
> ```
> docker run --rm -e 'VSCODE_PROXY_URI=https://some-{{port}}-domain.tld' -p 8080:8080 codercom/code-server --auth none
> ```
thanks
